pallets/unicks/src/kilt_unicks.rs

```diff
@@ -0,0 +1,159 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2022 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use sp_std::{fmt::Debug, marker::PhantomData, str, vec::Vec};
+
+use codec::{Decode, Encode};
+use frame_support::{ensure, sp_runtime::SaturatedConversion, traits::Get, BoundedVec};
+use scale_info::TypeInfo;
+
+use crate::{Config, Error};
+
+/// A KILT Unick.
+///
+/// It is bounded in size (inclusive range [MinLength, MaxLength]) and can only
+/// contain a subset of ASCII characters.
+#[derive(Encode, Decode, TypeInfo)]
+#[scale_info(skip_type_params(T, MaxLength, MinLength))]
+pub struct AsciiUnick<T, MinLength, MaxLength>(
+	pub(crate) BoundedVec<u8, MaxLength>,
+	PhantomData<T>,
+	PhantomData<MinLength>,
```

Comment:
Q: What do the entries 1 and 2 represent here? 0 seems to be `UnickInput`.

Reply:
It's phantom data to make it possible to use `impl<T: Config>` in the next block. This is needed to keep the `AsciiUnick` type generic.

Reply:
```suggestion
	pub(crate) BoundedVec<u8, MaxLength>,
	PhantomData<(T, MinLength)>,
```